android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiWebChromeClient.java

```diff
@@ -224,256 +387,263 @@ public boolean interceptOnBackPressed()
 		}
 
-		@Override
-		public Object call(KrollObject krollObject, Object[] args)
-		{
-			return null;
-		}
-
-		@Override
-		public void callAsync(KrollObject krollObject, HashMap args)
-		{
-			int resultCode = Activity.RESULT_CANCELED;
-			Object objectResults = args.get(TiC.EVENT_PROPERTY_RESULT_CODE);
-			if (objectResults instanceof Integer) {
-				resultCode = (Integer) objectResults;
+		// Set the intent's action.
+		String actionName = Intent.ACTION_GET_CONTENT;
+		if (chooserParams.isCaptureEnabled()) {
+			// Use camera photo/video taking action. (There is no action for capturing audio only.)
+			if (hasVideoMimeType || hasAudioMimeType) {
+				actionName = MediaStore.ACTION_VIDEO_CAPTURE;
+			} else {
+				actionName = MediaStore.ACTION_IMAGE_CAPTURE;
 			}
-			IntentProxy intentProxy = (IntentProxy) args.get(TiC.EVENT_PROPERTY_INTENT);
-			Intent data = null;
-			if (intentProxy != null) {
-				data = intentProxy.getIntent();
+		} else if ((acceptTypes.length > 1) && !hasImageMimeType && !hasVideoMimeType) {
+			// Only use this action when given multiple mime types, but none of them are photos/videos.
+			// Will show files from storage, but won't show files from photo gallery or camera's sandbox.
+			actionName = Intent.ACTION_OPEN_DOCUMENT;
+		}
+		intent.setAction(actionName);
+
+		// Set the mime type(s) of the files allowed to be chosen. Default to all types if not set.
+		if (!chooserParams.isCaptureEnabled()) {
+			intent.setType("*/*");
+			if (acceptTypes.length > 1) {
+				// Set multiple mime types. (Main type must be "*/*" or else this Extra will be ignored.)
+				intent.putExtra(Intent.EXTRA_MIME_TYPES, acceptTypes);
+			} else if (acceptTypes.length == 1) {
+				// Only given 1 mime type. Use it as the main type.
+				String mimeType = acceptTypes[0];
+				if ((mimeType != null) && !mimeType.isEmpty()) {
+					intent.setType(mimeType);
+				}
 			}
+		}
 
-			Uri results = null;
-			if (resultCode == Activity.RESULT_OK) {
-				if (data == null || (data.getDataString() == null || data.getDataString().isEmpty())) {
-					// If there is no data, then we may have taken a photo
-					if (mCameraPhotoUri != null) {
-						results = mCameraPhotoUri;
-					}
-				} else {
-					String dataString = data.getDataString();
-					if (dataString != null) {
-						results = Uri.parse(dataString);
-					}
-				}
+		// Set up camera capture handling if needed.
+		if (actionName.equals(MediaStore.ACTION_IMAGE_CAPTURE)) {
+			// Fetch app name and restrict it to chars: a-z, A-Z, 0-9, periods, dashes, and underscores
+			String normalizedAppName = TiApplication.getInstance().getAppInfo().getName();
+			normalizedAppName = normalizedAppName.replaceAll("[^\\w.-]", "_");
+
+			// Generate file name to be used under the system's "Pictures" directory.
+			String fileName
+				= TiApplication.getInstance().getAppInfo().getName().replaceAll("[^\\w.-]", "_")
+				+ "_"
+				+ (new SimpleDateFormat("yyyyMMdd_HHmmssSSS")).format(new Date());
```

Comment:
I didn't finish my code refactoring. The `normalizedAppName` variable was intended to be used in the string concatenation here.